### PR TITLE
feat(medusa): Add `metadata` to `StorePostCartsCartLineItemsItemReq`

### PR DESCRIPTION
**Pull request by erikengervall**
This is my first contribution.

I tried my best to follow the available documentation on how to contribute -- LMK if there's anything missing 🙏🏼 

## What - what changes are in this PR

This PR adds the Line Item `metadata` field to the `StorePostCartsCartLineItemsItemReq` request.

If omitted, no changes are applied.

## Why - why are these changes relevant

I created a [feature request](https://github.com/medusajs/medusa/discussions/4229) very recently and figured I'd take a stab at creating a PR myself. Here's a quote of the particular use case:

> We've got a use case where we'd like to enrich line items' `metadata` field just before entering the checkout flow. Without the addition of `metadata` to `StorePostCartsCartLineItemsItemReq`, the options I see are
> - Creating a custom endpoint that's almost identical to `StorePostCartsCartLineItemsItemReq` but that accepts `metadata`
>   - Although this would work just fine in the short term, we would effectively opt out of future updates to this endpoint, which I would prefer not to do.
> - Call `deleteItem` followed by `addItem` as `addItem` accepts `metadata`.
>   - This is quite hacky and it also causes the UI to render the Empty Cart page for a brief moment.

## How - how have the changes been implemented

Not sure what this refers to.

The changes were developed locally and pushed to a new branch on my fork.

## Testing - how has the changes been tested or how can the reviewer test the feature

Two additional unit tests have been created to cover this particular functionality.

### CODE REVIEW

File: packages/medusa/src/api/routes/store/carts/update-line-item.ts [hunk 1]
```diff
@@ -89,7 +89,7 @@ export default async (req, res) => {
         variant_id: existing.variant.id,
         region_id: cart.region_id,
         quantity: validated.quantity,
-        metadata: existing.metadata || {},
+        metadata: validated.metadata || existing.metadata || {},
```
Comment (olivermrbl) [inline packages/medusa/src/api/routes/store/carts/update-line-item.ts]: todo: I believe we can do `validated.metadata || {}` and let our dedicated [update metadata util](https://github.com/medusajs/medusa/blob/1304657133585f7850807d9508be2780a08d2b12/packages/medusa/src/utils/set-metadata.ts#L9) take care of this. The new and existing metadata will be merged correctly in the [`LineItemService.update` method](https://github.com/medusajs/medusa/blob/1304657133585f7850807d9508be2780a08d2b12/packages/medusa/src/services/line-item.ts#L451C6-L451C6).

Would be good with a test to verify

**Review by olivermrbl — COMMENTED** [review]
Thanks for the contribution @erikengervall. I have a minor suggestion. Let me know what you think :)

File: packages/medusa/src/api/routes/store/carts/update-line-item.ts (hunk 1 above)
Comment (olivermrbl) [reply packages/medusa/src/api/routes/store/carts/update-line-item.ts:92]: Note, the reason I am highlighting is, that the current implementation would overwrite any existing metadata.

**Review by olivermrbl — APPROVED** [review]
But otherwise LGTM, feel free to merge as you please :)

File: packages/medusa/src/api/routes/store/carts/update-line-item.ts (hunk 1 above)
Comment (erikengervall) [reply packages/medusa/src/api/routes/store/carts/update-line-item.ts]: Merging sounds even better 👍🏼 

Feel free to make the change, if not I’ll have a look tomorrow (it’s late here) 🙂

Comment (olivermrbl) [reply packages/medusa/src/api/routes/store/carts/update-line-item.ts]: All good - I'll push the changes so we can include it in today's release :)

**Pull request merged**